Lesson16/elifcetin/script.js

```diff
@@ -0,0 +1,456 @@
+  let count = 0;
+  for(let i = 0; i < arr.length; i++){
+  if(arr[i] === word){
+    ++count
+  }
+  }
+  console.log(count)
+}
+countOccurrences(["I", "love", "love", "you", "so much"], "love")
+
+/*
+19. Remove Falsy Values
+    - Define a function `removeFalsyValues(arr)` that loops through an array
+      and returns a new array without falsy values (false, 0, "", null, undefined, NaN).
+    - Log the new array.
+*/ function removeFalsyValues(arr){
+  newArr = [];
+  for(let i = 0; i < arr.length; i++){
+    if(arr[i]){
+      newArr.push(arr[i])
+    }
+
+    
+  }
+  console.log(newArr)
+} removeFalsyValues([0,1,2,3, null, NaN, "Hi"])
+
+/*
+20. Sum of All Digits in a String
+    - Define a function `sumDigits(str)` that loops through each character of `str`,
+      checks if it's a digit, and if so, adds it to a total sum.
+    - Log the final sum.
+    - Example: "abc123" -> 6
+*/  function sumDigits(str){
+  let sum = 0;
+  for ( let i = 0; i < str.length; i++){
+    if(!isNaN(str[i]) && str[i] !== " "){
+      sum = sum + Number(str[i])
+    }
+  }
+  console.log(sum)
+}
+sumDigits("hiiii78")
+
+/*
+21. Average of Array Elements
+    - Define a function `averageArray(numbers)` that uses a loop
+      to calculate the average (sum / length).
+    - Log the average.
+
+*/ function averageArray(numbers){
+  let sum = 0;
+  let average = 0;
+  for(let i = 0; i < numbers.length; i ++){
+    sum = sum + numbers[i] ;
+    average = sum / numbers[i]
+  }
+  
+  console.log(average)
+}
+
+averageArray([1,2,3,4,5,6,7])  
+
+/*
+22. Flatten a 2D Array (Nested Loops)
+    - Define a function `flattenArray(twoDArray)` that takes an array of arrays
+      (e.g., [[1,2],[3,4]]) and uses nested loops to create a new one-dimensional array.
+    - Log the flattened array.
+*/
+
+function flattenArray(twoDArray){
+  let flatArr = []
+  for(let i = 0; i < twoDArray.length; i++){
+    for(let j = 0; j < twoDArray[i].length; j++){
+      flatArr.push(twoDArray[i][j])
+    }
+  }
+  console.log(flatArr)
+ 
+}
+
+flattenArray([[1,2], [3,4], [5,6]])
+
+
+/*
+23. Find Words Containing a Letter
+    - Define a function `findWordsWithLetter(words, letter)` that loops through
+      an array of words and returns a new array of only the words that contain
+      the given letter.
+    - Log the filtered array.
+*/ function findWordsWithLetter(words, letter){
+  let result = []
+ 
+ 
+  for(let i = 0; i < words.length; i++){
+     if (words[i].includes(letter)){
+      result.push(words[i])
+     }
+
+  }
+  console.log(result)
+}
+
+findWordsWithLetter(["apple", "cherry", "break"], "r" )
+/*
+24. Push and Pop Operations
+    - Define a function `pushPopExample(arr, itemToPush)` that:
+      - pushes itemToPush to arr
+      - logs the updated array
+      - then pops the last element
+      - logs the popped element
+      - logs the final array
+*/ function pushPopExample(arr, itemToPush){
+   arr.push(itemToPush);
+   console.log(arr)
+   
+   return arr.pop();
```

Comment:
There's no requirement to add return statement in this function. Instead, you need to log the popped element and log the final array after `arr.pop();`